### PR TITLE
Handle LeftAssoc and RightAssoc in kore_to_kast

### PR DESCRIPTION
**Pull request by ehildenb**
This PR:

- Adds `Assoc.pattern` and defines it for `LeftAssoc` and `RightAssoc`, which resolves to the full associative pattern.
- Adds (previously failing) tests of `kore_to_kast` on a `LeftAssoc` and a `RightAssoc` pattern.
- Adds the ability for `KPrint.kore_to_kast` to convert `LeftAssoc` and `RightAssoc` patterns (using `Assoc.pattern`).

### CODE REVIEW

File: src/integration_tests/test_kore_to_kast.py [hunk 1]
```diff
@@ -227,3 +227,55 @@ def test_kast_to_kore(self) -> None:
...
+                    )
+                ),
+                KApply(
+                    '_Map_',
+                    [
+                        KApply('_Map_', [KVariable('X', sort=KSort('Map')), KVariable('Y', sort=KSort('Map'))]),
+                        KVariable('Z', sort=KSort('Map')),
+                    ],
+                ),
+            ),
```
Comment (ehildenb) [inline src/integration_tests/test_kore_to_kast.py:254]: I actually can't find any documentation if this is the correct associative order. That is, `LeftAssoc` means that parenthesis should be nested around left-more elements. @dwightguth can you confirm?

File: src/pyk/kore/syntax.py [hunk 2]
```diff
@@ -1345,6 +1350,15 @@ def let(self, *, app: Optional[App] = None) -> 'LeftAssoc':
     def map_pattern(self: 'LeftAssoc', f: Callable[[Pattern], Pattern]) -> 'LeftAssoc':
         return self
 
+    @property
+    def pattern(self) -> Pattern:
+        assert len(self.app.sorts) == 0
+        assert len(self.app.patterns) > 0
```
Comment (tothtamas28) [inline src/pyk/kore/syntax.py:1356]: I think these should raise `ValueError` instead.

File: src/pyk/kore/syntax.py [hunk 3]
```diff
@@ -1324,6 +1324,11 @@ class MLSyntaxSugar(MLPattern):
 class Assoc(MLSyntaxSugar):
     app: App
 
+    @property
+    @abstractmethod
+    def pattern(self) -> Pattern:
```
Comment (tothtamas28) [inline src/pyk/kore/syntax.py:1329]: What do you think about `desugared`?

**Review by tothtamas28 — APPROVED** [review]

File: src/pyk/kore/syntax.py (hunk 3 above)
Comment (ehildenb) [reply src/pyk/kore/syntax.py:1329]: I don't like `pattern` very much, but I think I like `desugared` even less though. We can change it later, or add a wrapper later.

File: src/pyk/kore/syntax.py (hunk 2 above)
Comment (ehildenb) [reply src/pyk/kore/syntax.py]: Done.

File: src/integration_tests/test_kore_to_kast.py (hunk 1 above)
Comment (ehildenb) [reply src/integration_tests/test_kore_to_kast.py:254]: @radumereuta perhaps you know?

Comment (dwightguth) [reply src/integration_tests/test_kore_to_kast.py:254]: Left associativity does indeed mean that nested elements occur under the left side.

**Pull request merged**